server/pulp/server/managers/consumer/applicability.py

```diff
@@ -107,6 +107,43 @@ def regenerate_applicability_for_consumers(consumer_criteria):
             manager.regenerate_applicability(profile_hash, content_type, profile_id, repo_id)
 
     @staticmethod
+    def regenerate_applicability_for_repos(repo_criteria):
+        """
+        Regenerate and save applicability data affected by given updated repositories.
+
+        :param repo_criteria: The repo selection criteria
+        :type repo_criteria: dict
+        """
+        repo_criteria = Criteria.from_dict(repo_criteria)
+
+        # Process repo criteria
+        repo_criteria.fields = ['id']
+        repo_ids = [r.repo_id for r in model.Repository.objects.find_by_criteria(repo_criteria)]
+
+        for repo_id in repo_ids:
+            # Find all existing applicabilities for given repo_id. Setting batch size of 5 ensures
+            # the MongoDB cursor does not time out. See https://pulp.plan.io/issues/998#note-6 for
+            # more details.
+            existing_applicabilities = RepoProfileApplicability.get_collection().find(
+                {'repo_id': repo_id}).batch_size(5)
+            for existing_applicability in existing_applicabilities:
+                existing_applicability = RepoProfileApplicability(**dict(existing_applicability))
+                profile_hash = existing_applicability['profile_hash']
+                unit_profile = UnitProfile.get_collection().find_one({'profile_hash': profile_hash},
+                                                                     fields=['id', 'content_type'])
```

Comment:
Denis when test i get the traceback http://fpaste.org/315282/04074145/
due to pymongo3 cnages you need to replace fields with projection